addons/sale/migrations/11.0.1.1/post-migration.py

```diff
@@ -8,9 +9,64 @@
 ]
 
 
+@openupgrade.logging()
+def fill_sale_order_line_computed(env):
+    """Compute through SQL sale.order.line fields for speeding up migration."""
+    # We first set everything to 0 in amt_invoiced
+    openupgrade.logged_query(
+        env.cr, "UPDATE sale_order_line sol SET amt_invoiced = 0.00",
+    )
+    # Then we set the quantity for the order lines with related invoice lines
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE sale_order_line sol
+        SET amt_invoiced = sub.amount
+        FROM (
+            SELECT rel.order_line_id, SUM(ail.price_total) AS amount
```

Comment:
Can you do a CASE switch to use the alternative amount in case of delivery products? https://github.com/odoo/odoo/blob/11.0/addons/sale/models/sale.py#L743

Reply:
now is line https://github.com/odoo/odoo/blob/11.0/addons/sale/models/sale.py#L763, and yes, it's done in https://github.com/OCA/OpenUpgrade/pull/1443/files#diff-2e401a61b8a9480b8995b85d4c21e9b0R43

marked as resolved